CMake/Modules/UrhoCommon.cmake

```diff
@@ -1106,13 +1107,13 @@ macro (add_html_shell)
             set (HTML_SHELL ${ARGN})
         else ()
             # Create Urho3D custom HTML shell that also embeds our own project logo
-            if (NOT EXISTS ${CMAKE_BINARY_DIR}/Source/shell.html)
+            if (NOT EXISTS ${CMAKE_SOURCE_DIR}/bin/shell.html)
                 file (READ ${EMSCRIPTEN_ROOT_PATH}/src/shell.html HTML_SHELL)
                 string (REPLACE "<!doctype html>" "<!-- This is a generated file. DO NOT EDIT!-->\n\n<!doctype html>" HTML_SHELL "${HTML_SHELL}")     # Stringify to preserve semicolons
                 string (REPLACE "<body>" "<body>\n<script>document.body.innerHTML=document.body.innerHTML.replace(/^#!.*\\n/, '');</script>\n<a href=\"https://urho3d.github.io\" title=\"Urho3D Homepage\"><img src=\"https://urho3d.github.io/assets/images/logo.png\" alt=\"link to https://urho3d.github.io\" height=\"80\" width=\"160\" /></a>\n" HTML_SHELL "${HTML_SHELL}")
-                file (WRITE ${CMAKE_BINARY_DIR}/Source/shell.html "${HTML_SHELL}")
+                file (WRITE ${CMAKE_SOURCE_DIR}/bin/shell.html "${HTML_SHELL}")
             endif ()
```

Comment:
If the html shell is provided out-of-the-box now in our source tree, instead of being generated on-the-fly in our build tree, then these lines can be removed all together.

Reply:
The idea here was to leave this just so it's easy to go back to legacy template by simly removing the new one.

Reply:
I see but if that is the case then I find it a little offending because the script tries to write into "source tree" directly and user will end up in dirty source tree afterward.

Reply:
Makes sense, will update the PR

Reply:
updated with different approach